Dropbear - a SSH2 server

Copyright (c) 2002,2003 Matt Johnston
All rights reserved.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE. */

Date: Nov 10 13:10:24 2016

From: a4f48fb07ea44bba99b96c22155137ee9a310385 Nov 22 16:46:35 2016 

Subject: [PATCH] RDK-16690: Adding verbose output.

	This is to debug the connection when needed.

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: Balaji Punnuru <balaji_punnuru@cable.comcast.com>
---

Index: dropbear-2014.63/options.h
===================================================================
--- dropbear-2014.63.orig/options.h
+++ dropbear-2014.63/options.h
@@ -66,6 +66,7 @@ much traffic. */
 
 #define ENABLE_SVR_LOCALTCPFWD
 #define ENABLE_SVR_REMOTETCPFWD
+#define DEBUG_TRACE
 
 /* Enable Authentication Agent Forwarding */
 #define ENABLE_SVR_AGENTFWD
Index: dropbear-2014.63/cli-session.c
===================================================================
--- dropbear-2014.63.orig/cli-session.c
+++ dropbear-2014.63/cli-session.c
@@ -182,6 +182,7 @@ static void recv_msg_service_accept(void
  * service, userauth and channel requests */
 static void cli_sessionloop() {
 
+	FILE *pidfile = NULL;
 	TRACE2(("enter cli_sessionloop"))
 
 	if (ses.lastpacket == 0) {
@@ -265,6 +266,12 @@ static void cli_sessionloop() {
 					dropbear_exit("Backgrounding failed: %d %s", 
 							errno, strerror(errno));
 				}
+                              
+                                pidfile = fopen("/var/tmp/rssh.pid", "w");
+                                if (pidfile) {
+                                   fprintf(pidfile, "%d\n", getpid());
+                                   fclose(pidfile);
+                                }
 			}
 			
 #ifdef ENABLE_CLI_NETCAT
